### DISCUSSION TITLE
React Testing Library failing test with code 'Font.register...'

### DISCUSSION THREAD

**Discussion by mdodge-ecgrow** [diegomura/react-pdf#2159] [Q&A]
I have several components that are creating PDF documents with this library with custom imported fonts.  I am registering the fonts like this:
```js
Font.register({
	family: 'PTSerif',
	format: 'truetype',
	src: PTSerif,
});

Font.register({
	family: 'Raleway',
	format: 'truetype',
	src: Raleway,
});
```

Well, I am starting to write unit tests for all my code using React Testing Library.  And on these components I am getting a failing test with the following error: `TypeError: Cannot read properties of undefined (reading 'register')` and it highlights the `Font.register` line.  Any ideas how I can get around this?

I did add this import to my test file and did not see any changes in the test results:
```js
import {
	Page,
	Text,
	View,
	Document,
	StyleSheet,
	Canvas,
	Font,
	Image,
} from '@react-pdf/renderer';
```

**Update**
So I just tried adding the 'Font' function to this snippet in my test file:
```js
jest.mock('@react-pdf/renderer', () => ({
	PDFViewer: jest.fn(() => null),
	Font: jest.fn(() => null),
}));
```
And my error changed to: `TypeError: _renderer.Font.register is not a function`.  I tried to change the mocked function to 'Font.register' but my IDE doesn't seem to like that syntax.

**Comment by Krzyku** [comment]
So Your implementation of `Font` is a function that returns `null`. 
This should be closer to what you're doing:
```js
jest.mock('@react-pdf/renderer', () => ({
  // ...
  Font: { register: () => null },
}));
```

For further information, You should visit [jest docs](https://jestjs.io/docs/mock-functions#mocking-modules) 
Personally, I would consider if unit tests with mocked PDF make sense.
